src/TestCase.php

```diff
@@ -2,64 +2,93 @@
+use Amp\Artax\Response;
+use Amp\Parallel\Sync\Lock;
+use Amp\Parallel\Sync\Semaphore;
+use Amp\Promise;
+use Asynit\Assert\Assertion;
 use Asynit\Assert\AssertWebCaseTrait;
-use Asynit\Runner\FutureHttp;
-use Asynit\Runner\FutureHttpPool;
-use Http\Client\HttpAsyncClient;
-use Http\Message\RequestFactory;
+use Http\Message\MessageFactory;
 use Psr\Http\Message\RequestInterface;
 
 class TestCase
 {
     use AssertWebCaseTrait;
 
-    /** @var RequestFactory */
-    private $requestFactory;
+    /** @var MessageFactory */
+    private $messageFactory;
 
-    /** @var FutureHttpPool */
-    private $futureHttpPool;
+    /** @var Semaphore */
+    private $semaphore;
 
-    final public function __construct(RequestFactory $requestFactory, FutureHttpPool $pool)
-    {
-        $this->requestFactory = $requestFactory;
-        $this->futureHttpPool = $pool;
-    }
+    /** @var Client */
+    private $client;
 
-    /**
-     * @return FutureHttpPool
-     */
-    public function getFutureHttpPool()
+    final public function __construct(MessageFactory $messageFactory, Semaphore $semaphore)
     {
-        return $this->futureHttpPool;
+        $this->messageFactory = $messageFactory;
+        $this->semaphore = $semaphore;
     }
 
     /**
      * Run before each test.
      *
      * Allow to set default services and context, and also decorate the http async client.
      *
-     * @param HttpAsyncClient $asyncClient
+     * @param Client $asyncClient
      *
-     * @return HttpAsyncClient
+     * @return Client
      */
-    public function setUp(HttpAsyncClient $asyncClient)
+    public function setUp(Client $asyncClient): Client
     {
         return $asyncClient;
     }
 
+    final public function initialize()
+    {
+        $this->client = $this->setUp(new DefaultClient());
+    }
+
     /**
      * Allow to test a rejection or a resolution of an async call.
      *
-     * @param RequestInterface $requestInterface
+     * @param RequestInterface $request
      *
-     * @return FutureHttp
+     * @return Promise
      */
-    final protected function sendRequest(RequestInterface $requestInterface)
+    final protected function sendRequest(RequestInterface $request): Promise
     {
-        $runner = new FutureHttp($requestInterface);
-        $this->futureHttpPool->add($runner);
+        $test = Assertion::$currentTest;
+        $promise = \Amp\call(function () use($request) {
+            $req = new \Amp\Artax\Request($request->getUri(), $request->getMethod());
+            $req = $req->withProtocolVersions([$request->getProtocolVersion()]);
+            $req = $req->withHeaders($request->getHeaders());
+            $req = $req->withBody((string) $request->getBody());
+
+            /** @var Lock $lock */
+            $lock = yield $this->semaphore->acquire();
+            /** @var Response $response */
+            $response = yield $this->client->request($req);
+            $content = yield $response->getBody()->read();
```

Comment:
This only reads the first chunk of the body. If you want to read the full body, use `$content = yield $response->getBody()`, which will automatically buffer the full body.